src/components/tile-details-panel/texture-section.tsx

```diff
@@ -0,0 +1,418 @@
+
+  if (!bufferId) {
+    console.error("Failed to create the buffer object"); // eslint-disable-line
+  }
+
+  gl.bindBuffer(gl.ARRAY_BUFFER, bufferId);
+  gl.enableVertexAttribArray(positionArrayLocation);
+  gl.bufferData(gl.ARRAY_BUFFER, data, gl.STATIC_DRAW);
+  gl.vertexAttribPointer(positionArrayLocation, 2, gl.FLOAT, false, 0, 0);
+};
+
+// TEXTURE SHADERS
+
+const vs = `
+precision highp float;
+
+attribute vec2 position;
+varying vec2 uv;
+
+void main() {
+  gl_Position = vec4(position, 0.0, 1.0);
+  uv = vec2(position.x * .5, -position.y * .5) + vec2(.5, .5);
+}
+`;
+
+const fs = `
+precision highp float;
+
+uniform sampler2D tex;
+varying vec2 uv;
+
+void main() {
+  gl_FragColor = vec4(texture2D(tex, uv).rgb, 1.);
+}
+`;
+
+const AREA_SIZE = 149;
+const PREVIEW_AREA_SIZE = 592;
+
+const drawBitmapTexture = async (
+  image: ImageData | HTMLCanvasElement,
+  maxAreaSize: number
+): Promise<{ url: string; width: number; height: number }> => {
+  const bitmap = await createImageBitmap(image);
+
+  const canvas = document.createElement("canvas");
+  const ctx = canvas.getContext("2d");
+  if (!ctx) {
+    throw new Error("No 2d context");
+  }
+  const imageWidth = image.width;
+  const imageHeight = image.height;
+
+  canvas.width = imageWidth;
+  canvas.height = imageHeight;
+
+  const imageSizeMax = imageWidth > imageHeight ? imageWidth : imageHeight;
+  const coeff = imageSizeMax / maxAreaSize;
+  const areaWidth = imageWidth / coeff;
+  const areaHeight = imageHeight / coeff;
+
+  // Position the image on the canvas (0, 0), and specify width and height of the image (areaWidth, areaHeight)
+  ctx.drawImage(bitmap, 0, 0, areaWidth, areaHeight);
+
+  return {
+    url: canvas.toDataURL("image/png"),
+    width: areaWidth,
+    height: areaHeight,
+  };
+};
+
+const drawCompressedTexture = async (
+  data,
+  maxAreaSize: number
+): Promise<{ url: string; width: number; height: number }> => {
+  const canvas = document.createElement("canvas");
+  const gl = canvas.getContext("webgl");
+  instrumentGLContext(gl);
+  if (!gl) {
+    throw new Error("No webgl context");
+  }
+
+  const program = new Program(gl, { vs, fs });
+  createAndFillBufferObject(gl, program);
+
+  const images = [data.data[0]]; // The first image only
+  const imageWidth = data.width;
+  const imageHeight = data.height;
+
+  canvas.width = imageWidth;
+  canvas.height = imageHeight;
+
+  gl.viewport(0, 0, imageWidth, imageHeight);
+  renderCompressedTexture(gl, program.handle, images);
+
+  return await drawBitmapTexture(canvas, maxAreaSize);
+};
+
+type TextureSectionProps = {
+  tile: Tile3D;
+};
+
+export const TextureSection = ({ tile }: TextureSectionProps) => {
+  const [texture, setTexture] = useState<string>("");
+  const [previewTexture, setPreviewTexture] = useState<string>("");
+  const [showPreviewTexture, setShowPreviewTexture] = useState<boolean>(false);
+
+  const [size, setSize] = useState<SIZE>({
+    width: AREA_SIZE,
```

Comment:
Confusing naming. "Area" is something equal to `width`*`height`

```suggestion
    width: SIZE,
```